### DISCUSSION TITLE
Impersonated account gas fee

### DISCUSSION THREAD

**Discussion by marioiordanov** [foundry-rs/foundry#7627] [Q&A]
When using an impersonated account, which account is being used to pay the gas fees

**Comment by Joel-eth** [comment]
the gas fees are typically paid by the user who initiated the transaction or by the smart contract being interacted with. Impersonating an account doesn't change the fundamental mechanics of how gas fees are handled in a blockchain transaction. The gas fees are still deducted from the user's wallet or account balance.
